packages/material-ui/src/ButtonBase/ButtonBase.js

```diff
@@ -470,10 +474,9 @@ ButtonBase.propTypes = {
    */
   TouchRippleProps: PropTypes.object,
   /**
-   * Used to control the button's purpose.
-   * This prop passes the value to the `type` attribute of the native button component.
+   * @ignore
```

Comment:
With our types it would require more work to get JSDOC working. `type` is only considered when `component="button"`. It has different types depending on the `component` so we can't just slap it into props with `type: React.ButtonHTMLAttributes['type']`.

It was also to narrow previously. I'd say this is fine since the documentation value is pretty low. I doubt this is only used after reading about it here.